src/components/button/button.spec.js

```diff
@@ -0,0 +1,364 @@
+
+    Object.keys(sizes).forEach((size) => {
+      variants.forEach((variant) => {
+        it(`matches the style for a ${size} ${variant} Button`, () => {
+          const wrapper = render({
+            children: 'foo', disabled: true, renderAs: variant, size
+          }, TestRenderer.create).toJSON();
+
+          assertStyleMatch({
+            background:
+            (variant === 'secondary' || variant === 'tertiary' ? 'transparent' : BaseTheme.disabled.button),
+            borderColor: (variant === 'secondary' ? BaseTheme.disabled.button : 'transparent'),
+            color: BaseTheme.disabled.text,
+            fontSize: (size === 'large' ? '16px' : '14px'),
+            height: `${sizes[size][0].toString()}px`,
+            paddingLeft: `${sizes[size][1].toString()}px`,
+            paddingRight: `${sizes[size][1].toString()}px`
+          }, wrapper);
+        });
+      });
+    });
+  });
+
+  describe('when the classic theme is applied', () => {
+    ['blue', 'grey', 'magenta', 'magenta-dull', 'red', 'white'].forEach((variant) => {
+      const wrapper = TestRenderer.create(
+        <StyledButton theme={ classicTheme } variant={ variant }>Foo</StyledButton>
+      );
+
+      it(`matches the snapshot for the ${variant} Button when default props are passed`, () => {
+        expect(wrapper).toMatchSnapshot();
+      });
+    });
+
+    it('matches the expected style default "blue" Button default props are passed', () => {
+      const wrapper = TestRenderer.create(<StyledButton theme={ classicTheme }>foo</StyledButton>);
+      assertStyleMatch({
+        background: 'transparent',
+        border: '1px solid #255bc7',
+        color: '#255bc7'
+      }, wrapper.toJSON());
+    });
+
+    it('matches the expected style default "blue" Button default props are passed', () => {
+      const wrapper = TestRenderer.create(<StyledButton disabled theme={ classicTheme }>foo</StyledButton>);
+      assertStyleMatch({
+        background: '#e6ebed'
+      }, wrapper.toJSON());
+    });
+  });
+
+  let defaultButton, primary, secondary, small, large, disabled, anchor, to;
+
+  beforeEach(() => {
+    defaultButton = render({ children: 'Save' });
+
+    primary = render({
+      name: 'Primary Button',
+      as: 'primary',
+      onClick: jest.fn(),
+      children: 'Primary'
+    });
+
+    secondary = render({
+      name: 'Secondary Button',
+      className: 'customClass',
+      theme: 'red',
+      children: 'Secondary'
+    });
+
+    small = render({
+      name: 'Small Button',
+      size: 'small',
+      children: 'Small'
+    });
+
+    large = render({
+      name: 'Large Button',
+      size: 'large',
+      children: 'Large'
+    });
+
+    disabled = render({
+      name: 'Disabled Button',
+      disabled: true,
+      children: 'Disabled'
+    });
+
+    anchor = render({
+      href: '/foo',
+      children: 'Anchor'
+    }, mount);
+
+    to = render({
+      to: '/foo',
+      children: 'To'
+    }, mount);
+  });
```

Comment:
I think it would be better to have each test with a separate data

Reply:
These are the tests remaining from the legacy button so I just left them as is wherever possible as it demonstrates that the backwards compatibility is maintained

Reply:
don't you think that these tests should be rewritten, or removed when unnecessary?

Reply:
I want to try and leave them be wherever possible as it is clearer in demonstrating the backwards compatibility, I have refactored it so each one is in its own `describe` as it is odd the way it is